crates/state/src/da_blob.rs

```diff
@@ -83,49 +117,49 @@ impl BlobSpec {
 ///
 /// These are never stored on-chain.
 #[derive(Clone, Debug, Eq, PartialEq, Arbitrary, BorshDeserialize, BorshSerialize)]
-pub struct BlobIntent {
-    /// The destination for this blob.
-    dest: BlobDest,
+pub struct PayloadIntent {
+    /// The destination for this payload.
+    dest: PayloadDest,
 
-    /// Commitment to the blob payload.
+    /// Commitment to the payload.
     commitment: Buf32,
 
     /// Blob payload.
-    payload: Vec<u8>,
+    payload: L1Payload,
 }
 
-impl BlobIntent {
-    pub fn new(dest: BlobDest, commitment: Buf32, payload: Vec<u8>) -> Self {
+impl PayloadIntent {
+    pub fn new(dest: PayloadDest, commitment: Buf32, payload: L1Payload) -> Self {
         Self {
             dest,
             commitment,
             payload,
         }
     }
 
-    /// The target we expect the DA blob to be stored on.
-    pub fn dest(&self) -> BlobDest {
+    /// The target we expect the DA payload to be stored on.
+    pub fn dest(&self) -> PayloadDest {
         self.dest
     }
 
-    /// Commitment to the blob payload, which might be context-specific.  This
-    /// is conceptually unrelated to the blob ID that we use for tracking which
-    /// blobs we've written in the L1 writer bookkeeping.
+    /// Commitment to the payload, which might be context-specific.  This
+    /// is conceptually unrelated to the payload ID that we use for tracking which
+    /// payloads we've written in the L1 writer bookkeeping.
     pub fn commitment(&self) -> &Buf32 {
         &self.commitment
     }
 
-    /// The blob payload that matches the commitment.
-    pub fn payload(&self) -> &[u8] {
+    /// The payload that matches the commitment.
+    pub fn payload(&self) -> &L1Payload {
         &self.payload
     }
 
-    /// Generates the spec from the relevant parts of the blob intent that
-    /// uniquely refers to the blob data.
-    pub fn to_spec(&self) -> BlobSpec {
-        BlobSpec {
+    /// Generates the spec from the relevant parts of the payload intent that
+    /// uniquely refers to the payload data.
+    pub fn to_spec(&self) -> PayloadSpec {
+        PayloadSpec {
```

Comment:
`::new`?

Reply:
It is in the same module so doesn't really matter much imo.

Reply:
If there was no public facing function then sure, but there is.

Reply:
I've added a `::new` method.